doc/dev.md

```diff
@@ -109,38 +120,33 @@ Return generic "access denied" error message to user.
+Errors are being logged at `QuickFeed Service` level. All other methods called from there (including database and SCM methods) will just wrap and return all error messages directly. Introduce logging on layers deeper than `QuickFeed Service` only if necessary.
 
-Errors returned to user interface must be few and informative, yet should not provide too many information about server routines. User must only be informed about details he or she can do something about.
+Errors returned to a user should be few and informative. They should not reveal internal details of the application.
 
 ### Frontend
 
-When receiving response from the server, response status code is checked on the frontend. Any message with code different from 0 (0 is gRPC status code `OK`) will be logged to console. Relevant error messages will be displayed to user on course and group creation, and user and group enrollment updates.
+When receiving a response from the server, the response status code is checked on the frontend. Any message with code different from 0 (0 is status code `OK`) will be logged to console. Error messages will be displayed to user where relevant, e.g. on course and group creation, and user and group enrollment updates.
 
-[gRPC status codes](https://github.com/grpc/grpc/blob/master/doc/statuscodes.md)
+[Connect Error Codes](https://connectrpc.com/docs/protocol#error-codes)
 
 ## GitHub API
 
-For GitHub integration we are using [Go implementation](https://github.com/google/go-github/tree/master/github) of [GitHub API](https://developer.github.com/v3/)
+For GitHub integration we are using [Go implementation](https://github.com/google/go-github) of [GitHub API](https://docs.github.com/en/rest)
 
 ### Webhooks
 
-- GitHub [Webhooks API](https://developer.github.com/webhooks/) is used for building and testing of code submitted by students.
-- webhook is created automatically on course creation. It will react to every push event to any of course organization's repositories.
-- depending on the repository the push event is coming from, assignment information will be updated in the QuickFeed's database, or a docker container with a student solution code will be built
-- `name` field for any GitHub webhook is always "web"
-- webhook will be using the same callback URL you have provided to the QuickFeed OAuth2 application and in the server startup command
+- GitHub [Webhooks API](https://docs.github.com/en/webhooks) is used for building and testing of code submitted by students.
+- A webhook is created automatically when installing the GitHub App on a course organization. It will react to every push event to any of course organization's repositories.
+- depending on the repository the push event is coming from, assignment information will be updated in the QuickFeed's database, or a docker container with a student solution code will be built and tested.
```

Comment:
**D**epending on **which repository** ... updated in --the-- QuickFeed's database.

I felt this sentence wasn't so clear. Suggest splitting this point into separate sentences, like:
- Push events from the `tests` repository may update
  - The assignment information in QuickFeed's database.
  - The docker container and run.sh script used for building and testing student submitted code.
- Push events from `username-labs` repositories may trigger test execution.